src/wp-includes/class-wp-duotone.php

```diff
@@ -948,45 +940,41 @@ public static function register_duotone_support( $block_type ) {
 	 * This handles selectors defined in `color.__experimentalDuotone` support
 	 * if `filter.duotone` support is not defined.
 	 *
-	 * @param string $block_name The block name.
+	 * @param WP_Block_Type $block_type Block type to check for support.
 	 *
 	 * @internal
 	 *
 	 * @since 6.3.0
 	 *
 	 * @return string The CSS selector or null if there is no support.
 	 */
-	private static function get_selector( $block_name ) {
-		$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block_name );
-
-		if ( $block_type && property_exists( $block_type, 'supports' ) ) {
-			/*
-			 * Backwards compatibility with `supports.color.__experimentalDuotone`
-			 * is provided via the `block_type_metadata_settings` filter. If
-			 * `supports.filter.duotone` has not been set and the experimental
-			 * property has been, the experimental property value is copied into
-			 * `supports.filter.duotone`.
-			 */
-			$duotone_support = _wp_array_get( $block_type->supports, array( 'filter', 'duotone' ), false );
-			if ( ! $duotone_support ) {
-				return null;
-			}
-
-			/*
-			 * If the experimental duotone support was set, that value is to be
-			 * treated as a selector and requires scoping.
-			 */
-			$experimental_duotone = _wp_array_get( $block_type->supports, array( 'color', '__experimentalDuotone' ), false );
-			if ( $experimental_duotone ) {
-				$root_selector = wp_get_block_css_selector( $block_type );
-				return is_string( $experimental_duotone )
-					? WP_Theme_JSON::scope_selector( $root_selector, $experimental_duotone )
-					: $root_selector;
-			}
+	private static function get_selector( $block_type ) {
+		/*
+		 * Backwards compatibility with `supports.color.__experimentalDuotone`
+		 * is provided via the `block_type_metadata_settings` filter. If
+		 * `supports.filter.duotone` has not been set and the experimental
+		 * property has been, the experimental property value is copied into
+		 * `supports.filter.duotone`.
+		 */
+		$duotone_support = block_has_support( $block_type, array( 'filter', 'duotone' ) );
+		if ( ! $duotone_support ) {
+			return null;
+		}
 
-			// Regular filter.duotone support uses filter.duotone selectors with fallbacks.
-			return wp_get_block_css_selector( $block_type, array( 'filter', 'duotone' ), true );
+		/*
+		 * If the experimental duotone support was set, that value is to be
+		 * treated as a selector and requires scoping.
+		 */
+		$experimental_duotone = block_has_support( $block_type, array( 'color', '__experimentalDuotone' ) );
```

Comment:
I suspect the problem with double filters is this line. `block_has_support` looks like it only returns a boolean rather than the value of the supports.

```suggestion
		$experimental_duotone = _wp_array_get( $block_type->supports, array( 'color', '__experimentalDuotone' ), false );
```

Reply:
Using `block_has_support` is cleaner here. It is easier to understand what is going on.

Reply:
We aren't just checking if it has support, we're also getting the string value of the support for experimental duotone. Using `block_has_support` is causing the bug.

Reply:
(We really should add tests for this and do better at writing these from the beginning, would be good follow-up task.)